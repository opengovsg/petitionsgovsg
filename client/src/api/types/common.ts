export type MessageResponse = {
  message: string
  data: unknown
}
