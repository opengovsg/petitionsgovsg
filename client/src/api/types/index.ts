export * from './post'
export * from './signature'
export * from './addressee'
export * from './auth'
