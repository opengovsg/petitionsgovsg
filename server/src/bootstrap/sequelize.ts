import { Sequelize } from 'sequelize'

import {
  defineSignature,
  definePost,
  defineAddressee,
  defineSubscription,
} from '@/models'
import { dbConfig } from './config/database'

export const sequelize = new Sequelize({ ...dbConfig, logging: false })

export const { Addressee } = defineAddressee(sequelize)
export const { Post } = definePost(sequelize, { Addressee })
export const Signature = defineSignature(sequelize, { Post })
export const Subscription = defineSubscription(sequelize, { Post })

export default sequelize
