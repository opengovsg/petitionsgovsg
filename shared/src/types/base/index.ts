export * from './signature'
export * from './post'
export * from './user'
export * from './addressee'
