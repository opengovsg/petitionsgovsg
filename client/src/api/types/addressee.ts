import { BaseModelParams, MessageResponse } from './common'

export type BaseAddresseeDto = BaseModelParams & {
  name: string
  shortName: string | null
}

export type CreateAddresseeReqDto = {
  name: string
  shortName: string | null
}

// Returns signature ID
export type CreateAddresseeResDto = MessageResponse & { data: number }
