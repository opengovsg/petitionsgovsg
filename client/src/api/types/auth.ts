import { BaseModelParams } from './common'

export type BaseUserNameDto = BaseModelParams & {
  fullname: string
}
