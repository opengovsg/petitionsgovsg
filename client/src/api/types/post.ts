import { BaseModelParams, MessageResponse } from './common'
import { PostStatus } from '~shared/types/base'
import { BaseSignatureDto } from './signature'

export type BasePostDto = BaseModelParams & {
  id: number
  userId: number
  title: string
  summary: string
  reason: string | null
  request: string | null
  references: string | null
  status: PostStatus
  fullname: string
  salt: string
}

// Backend does not select updatedAt
export type GetSinglePostDto = BasePostDto & {
  signatures: Pick<BaseSignatureDto, 'comment' | 'createdAt' | 'fullname'>[]
  signatureCount: number
}

export type GetPostsDto = {
  posts: GetSinglePostDto[]
  totalItems: number
}

export type CreatePostReqDto = Pick<
  BasePostDto,
  'title' | 'summary' | 'reason' | 'request' | 'references'
>

export type CreatePostResDto = MessageResponse & { data: number }

export type UpdatePostReqDto = CreatePostReqDto

export type UpdatePostResDto = MessageResponse
