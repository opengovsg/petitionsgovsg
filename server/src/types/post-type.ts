import { Post } from '~shared/types/base'

export type PostEditType = {
  id: number
  userid: number
  summary: string
  title: string
}

export type UpdatePostRequestDto = Pick<Post, 'title' | 'summary'>
