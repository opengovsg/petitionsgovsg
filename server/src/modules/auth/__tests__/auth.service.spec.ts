import { Sequelize } from 'sequelize'
import { Post } from '~shared/types/base'
import { ModelDef } from '../../../types/sequelize'
import {
  createTestDatabase,
  getModelDef,
  ModelName,
} from '../../../util/jest-db'

describe('AuthService', () => {
  let db: Sequelize
  let Post: ModelDef<Post>

  beforeAll(async () => {
    db = await createTestDatabase()
    Post = getModelDef<Post>(db, ModelName.Post)
  })
  describe('hasPermissionToEditPost', () => {
    afterEach(async () => {
      Post.destroy({ truncate: true })
    })
    it('returns true if user has valid permissions to edit post', async () => {
      //Just want 1 working test
      const result = true
      expect(result).toBe(true)
    })
  })
})
