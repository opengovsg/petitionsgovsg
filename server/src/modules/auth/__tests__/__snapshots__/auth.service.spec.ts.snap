// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthService verifyUserCanViewPost throws on archived post without userId 1`] = `"User must be logged in to access private post"`;
